src/eval_framework/utils/file_ops.py

```diff
@@ -149,35 +170,61 @@ def find_hf_checkpoint_root_from_path_list(self) -> Path | None:
         Returns:
             str | None: Path to the HuggingFace checkpoint root folder, or None if not found
         """
-        download_path = (
-            self.download_path.name
-            if isinstance(self.download_path, tempfile.TemporaryDirectory)
-            else self.download_path
-        )
-
-        if not download_path:
-            return None
 
-        checkpoint_roots = [x for x in Path(download_path).glob("**/config.json")]
-        if checkpoint_roots:
-            assert len(checkpoint_roots) == 1, (
-                "Multiple checkpoints found"
-            )  # if there are more than one, we have a problem
-            return checkpoint_roots[0].parent
+        # self.download_path can be None if download_artifact was never called
+        if self.download_path and self.download_path.exists():
+            checkpoint_roots = [x for x in Path(self.download_path).glob("**/config.json")]
+            if checkpoint_roots:
+                assert len(checkpoint_roots) == 1, (
+                    "Multiple checkpoints found"
+                )  # if there are more than one, we have a problem
+                return checkpoint_roots[0].parent
 
         return None
 
-    def cleanup(self) -> None:
-        self._cleanup_temp_dir()
-
     def __enter__(self) -> "WandbFs":
+        # Store original handlers for restoration
+        self._original_sigterm = signal.signal(signal.SIGTERM, self._clean_on_signal)
+        self._original_sigint = signal.signal(signal.SIGINT, self._clean_on_signal)
+
+        # on interrupt or termination
+        signal.signal(signal.SIGTERM, self._clean_on_signal)
+        signal.signal(signal.SIGINT, self._clean_on_signal)
+
         return self
 
     def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
-        self._cleanup_temp_dir()
+        """
+        exit the context manager, cleaning up the temporary directory if it was used
+        or the user directory if it was specified and failed to download.
+        """
+        if self.user_supplied_download_path:
+            self._cleanup_user_dir()
+        else:
+            self._cleanup_temp_dir()
+
+        signal.signal(signal.SIGTERM, self._original_sigterm)
+        signal.signal(signal.SIGINT, self._original_sigint)
+
+    def _cleanup_user_dir(self) -> None:
+        """
+        _cleanup_user_dir will remove the contents of the user specified
+        download path if there was an attempt to download the artifact and it failed.
+        """
+        if (
+            # check to make sure this flag was set; if not, there was no attempt to download.
+            hasattr(self, "_artifact_downloaded")
```

Comment:
suggestion: `if not getattr(self, "_artifact_downloaded", True) and self.user_supplied_download_path and ...`